src/org/jitsi/impl/neomedia/rtp/remotebitrateestimator/RemoteBitrateEstimatorAbsSendTime.java

```diff
@@ -0,0 +1,627 @@
+        this.incomingBitrate = new RateStatistics(kBitrateWindowMs,8000);
+        this.incomingBitrateInitialized = false;
+        this.totalProbesReceived = 0;
+        this.firstPacketTimeMs = -1;
+        this.lastUpdateMs = -1;
+        this.umaRecorded = false;
+       // RTC_DCHECK(observer_);
+        logger.warn("RemoteBitrateEstimatorAbsSendTime: Instantiating.");
+        //network_thread_.DetachFromThread();
+    }
+
+    List<Cluster> computeClusters(List<Cluster> clusters)
+    {
+        Cluster current = new Cluster();
+        long prevSendTime =  -1;
+        long prevRecvTime = -1;
+        for(Probe probe : probes_)
+        {
+            if(prevSendTime >= 0)
+            {
+                long sendDeltaMs = probe.sendTimeMs - prevSendTime;
+                long recvDeltaMs = probe.recvTimeMs - prevRecvTime;
+
+                if(sendDeltaMs >= 1 && recvDeltaMs >= 1)
+                {
+                    ++current.numAboveMinDelta;
+                }
+
+                if(!IsWithinClusterBounds(sendDeltaMs,current))
+                {
+                    if(current.count >= kMinClusterSize)
+                    {
+                        clusters = addCluster(clusters,current);
+                    }
+                    current = new Cluster();
+                }
+                current.sendMeanMs += sendDeltaMs;
+                current.recvMeanMs += recvDeltaMs;
+                current.meanSize += probe.payloadSize;
+                ++current.count;
+            }
+
+            prevSendTime = probe.sendTimeMs;
+            prevRecvTime = probe.recvTimeMs;
+        }
+        if(current.count >= kMinClusterSize)
+            clusters = addCluster(clusters, current);
+        return clusters;
+    }
+
+    /**
+     * @param clusters
+     * @returns a cluster that shows the best probe
+     */
+    Cluster findBestProbe(List<Cluster> clusters)
+    {
+        int highestProbeBitrateBps = 0;
+        Cluster bestIt = new Cluster();
+        for (Cluster cluster : clusters) {
+            if (cluster.sendMeanMs == 0 || cluster.recvMeanMs == 0)
+                continue;
+            if (cluster.numAboveMinDelta > cluster.count / 2 &&
+                    (cluster.recvMeanMs - cluster.sendMeanMs <= 2.0f &&
+                            cluster.sendMeanMs - cluster.recvMeanMs <= 5.0f))
+            {
+                int probeBitrateBps =
+                        Math.min(cluster.getSendBitrateBps(), cluster
+                                .getRecvBitrateBps());
+                if (probeBitrateBps > highestProbeBitrateBps)
+                {
+                    highestProbeBitrateBps = probeBitrateBps;
+                    bestIt = cluster;
+                }
+            }
+            else
+            {
+                /**
+                 * @Todo
+                 * What do we loose by converting from double to long ?
+                 */
+                double sendBitrateBps = cluster.meanSize * 8 * 1000
```

Comment:
Looks like the worst that could happen is that the values in the log message are not precise